render/gles2/surface.c

```diff
@@ -34,6 +34,35 @@ static bool gles2_surface_attach_pixels(struct wlr_surface_state *surface,
 	return true;
 }
 
+static bool gles2_surface_attach_shm(struct wlr_surface_state *surface,
+		uint32_t format, struct wl_shm_buffer *buffer) {
+	const struct pixel_format *fmt = gl_format_for_wl_format(format);
+	if (!fmt || !fmt->gl_format) {
+		wlr_log(L_ERROR, "No supported pixel format for this surface");
+		return false;
+	}
+	wl_shm_buffer_begin_access(buffer);
+	uint8_t *pixels = wl_shm_buffer_get_data(buffer);
+	int width = wl_shm_buffer_get_width(buffer);
+	int height = wl_shm_buffer_get_height(buffer);
+	int pitch = wl_shm_buffer_get_stride(buffer) / (fmt->bpp / 8);
```

Comment:
Weston calls this variable `pitch` although I don't know if that term is the same as `stride`. This division I added is necessary for setting `GL_UNPACK_ROW_LENGTH_EXT` because otherwise I get memory errors from stride being too large and the opengl skipping too many pixels.

Wayland seems to include some sanity checks for us so the stride will never cause these memory errors as long as we do this calculation.